samples/wifi/provisioning/implementation_details.rst

```diff
@@ -0,0 +1,195 @@
+.. _wifi_provisioning_details:
+
+Implementation
+##############
+
+.. contents::
+   :local:
+   :depth: 2
+
+This page explains the implementation of the Wi-Fi Provisioning Service sample.
+
+Task and event handling
+=======================
+
+The Wi-Fi Provisioning Service sample uses `Protocol Buffers`_ to define platform-independent message formats.
+These definitions are available as `.proto` files in the sample code.
+The sample also uses `nanopb`_ to convert encoded platform-independent messages to and from C.
+The device performs actions according to messages it exchanges with a configurator (the peer that sends the message, such as a smartphone running the nRF Wi-Fi Provisioner app).
```

Comment:
Can we add a link to the Requirements header of the description page with "nRF Wi-Fi Provisioner app" as the anchor?